vtex-apps/minicart
# Make minicart content more flexible

#### What is the purpose of this pull request?

- Make the `BaseContent` component render any children blocks passed to it.
- Create new interface `minicart-checkout-button` and use `sticky-layout` to make `minicart.v2` much more flexible.
- Update default blocks implementation.

#### What problem is this solving?

This should enable users to further customize the minicart. Enabling developers to do something like this https://app.clubhouse.io/vtex/story/29613/minicart-v2-should-be-more-flexible .

#### How should this be manually tested?

Go to [Workspace](https://minicartchildren--storecomponents.myvtex.com/), add items into the minicart and see that right above the minicart-summary there's a `flex-layout` block. This `flex-layout` was placed there using:

```json
"minicart.v2": {
    "children": ["minicart-base-content"]
  },
  "minicart-base-content": {
    "blocks": ["minicart-empty-state"],
    "children": ["minicart-product-list", "sticky-layout#minicart-footer"]
  },
  "sticky-layout#minicart-footer": {
    "children": [
      "flex-layout.row#deals",
      "minicart-summary",
      "minicart-checkout-button"
    ]
  },
  "minicart-product-list": {
    "blocks": ["product-list#minicart"]
  },
  "product-list#minicart": {
    "blocks": ["product-list-content-mobile"]
  },
  "minicart-summary": {
    "blocks": ["checkout-summary.compact#minicart"]
  },
(...)
```

#### Screenshots or example usage

![Screen Shot 2020-01-22 at 10 00 59](https://user-images.githubusercontent.com/27777263/72896376-6a3d8d80-3cfe-11ea-8c73-0c0602de5680.png)


#### Types of changes
